lib/tests/test_pathutils.c

```diff
@@ -88,3 +88,89 @@ Test(test_pathutils, test_get_filename_extension)
   cr_assert_null(get_filename_extension(""), "wrong file name extension returned");
   cr_assert_null(get_filename_extension(NULL), "wrong file name extension returned");
 }
+
+void
+test_get_filenames_setup(void)
+{
+  const gchar *dirs[] =
+    {
+      "pathutils_get_filenames",
+      "pathutils_get_filenames/testdir",
+      "pathutils_get_filenames/testdir2"
+    };
+  const gchar *files[] =
+    {
+      "pathutils_get_filenames/test.file",
+      "pathutils_get_filenames/test2.file",
+      "pathutils_get_filenames/testdir/test.file",
+      "pathutils_get_filenames/testdir2/test23.file",
+      "pathutils_get_filenames/testdir2/test22.file"
+    };
+  size_t dirs_len = sizeof(dirs) / sizeof(gchar *);
+  size_t files_len = sizeof(files) / sizeof(gchar *);
+
+  for (gint i = 0; i < dirs_len; ++i)
+    {
+      int status = mkdir(dirs[i], S_IRUSR | S_IWUSR | S_IXUSR);
+      cr_assert_not(status < 0, "mkdir error %s", dirs[i]);
+    }
+
+  for (gint i = 0; i < files_len; ++i)
+    {
+      int fd = open(files[i], O_CREAT | O_RDWR, 0644);
+      cr_assert_not(fd < 0, "open error %s", files[i]);
+      ssize_t done = write(fd, "a", 1);
+      cr_assert_eq(done, 1, "write error %s", files[i]);
+      int error = close(fd);
+      cr_assert_not(error, "close error %s", files[i]);
+    }
+}
+
+void
+test_get_filenames_teardown(void)
+{
+  const gchar *dirs[] =
+    {
+      "pathutils_get_filenames",
+      "pathutils_get_filenames/testdir",
+      "pathutils_get_filenames/testdir2"
+    };
+  const gchar *files[] =
+    {
+      "pathutils_get_filenames/test.file",
+      "pathutils_get_filenames/test2.file",
+      "pathutils_get_filenames/testdir/test.file",
+      "pathutils_get_filenames/testdir2/test23.file",
+      "pathutils_get_filenames/testdir2/test22.file"
+    };
+  size_t dirs_len = sizeof(dirs) / sizeof(gchar *);
+  size_t files_len = sizeof(files) / sizeof(gchar *);
+
+  for (gint i = 0; i < files_len; ++i)
```

Comment:
[g_remove](https://developer.gnome.org/glib/stable/glib-File-Utilities.html#g-remove) theoretically removes both files and directories.

Reply:
Thanks, fixed :)

Reply:
I was thinking this could be used to recursively delete files, but it seems it is not the case. Sorry for the extra round, the original version was fine.